File: docs/evidence/2026-09-25-rust-core-spikes/wasm/jwsbench.mjs
import { readFileSync } from 'node:fs';
const F = '../../../../fixtures/';
const root = new Uint8Array(readFileSync(F + 'generated/jws-root.der'));
const jws = readFileSync(F + 'generated/transaction.jws', 'utf8').trim();
const n = 500;
{ const { JwsVerifier } = await import('./pkg-nodejs/aprv_wasm.js');
  const v = new JwsVerifier('com.example.app', ['Sandbox'], [root]);
  for (let i = 0; i < 50; i++) v.verifyTransaction(jws);
  const t = performance.now(); for (let i = 0; i < n; i++) v.verifyTransaction(jws);
  console.log('wasm jws', ((performance.now() - t) / n * 1000).toFixed(0), 'us/op'); }
for (const entry of ['apple-purchase-receipt-verifier', 'apple-purchase-receipt-verifier/web']) {
  const m = await import(entry.endsWith('/web') ? './node_modules/apple-purchase-receipt-verifier/dist/web/index.js' : './node_modules/apple-purchase-receipt-verifier/dist/index.js');
  const v = new m.JwsVerifier({ trustedRoots: [entry.endsWith('/web') ? root : Buffer.from(root)], bundleId: 'com.example.app', acceptedEnvironments: ['Sandbox'] });
  for (let i = 0; i < 50; i++) await v.verifyTransaction(jws);
  const t = performance.now(); for (let i = 0; i < n; i++) await v.verifyTransaction(jws);
  console.log(entry, 'jws', ((performance.now() - t) / n * 1000).toFixed(0), 'us/op');
}

File: docs/evidence/2026-09-25-rust-core-spikes/uniffi/jws_bench.py
import time
from aprv_uniffi import JwsVerifier, Environment
F="../../../../fixtures/"
j = JwsVerifier("com.example.app", [Environment.SANDBOX], None, [open(F+"generated/jws-root.der","rb").read()])
s = open(F+"generated/transaction.jws").read().strip()
for _ in range(50): j.verify_transaction(s)
t=time.perf_counter(); n=500
for _ in range(n): j.verify_transaction(s)
print("native (uniffi/python) jws", round((time.perf_counter()-t)/n*1e6), "us/op")
